### PR TITLE
Issue/160.2 into Issue/160

### PR DESCRIPTION
**Pull request by ravenmaster001**
Changes from me learning the test suite while trying to get it to a working state. 
Only merges back into Issue/160 branch.

Addresses issues #160 #173 #192 

Also allows me to check travis PR specific scripts are working.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -9,6 +9,7 @@ exclude =
 
     # These files are mostly or completely autogenerated
     migrations,
+    manage.py,
```
Comment (hayleyavw) [inline .flake8]: is this file not covered by the line below?

File: .travis.yml [hunk 2]
```diff
@@ -8,10 +8,22 @@ python:
...
+before_script:
+  - cd csunplugged/
+  - chmod +x manage.py
 
 # Runs test suite
-script: flake8
+script:
+  - flake8 --config=../.flake8
+  - coverage run --rcfile=../.coveragerc manage.py test --pattern "test_*.py" -v 3
+  - 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then ./manage.py test --pattern "test_*.py" --reverse -v 0; fi'
```
Comment (hayleyavw) [inline .travis.yml]: is my interpretation correct: if the Travis' tests haven't failed up to this point, then run the test suite backwards

File: csunplugged/tests/BaseTestWithDB.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    @classmethod
+    def tearDownClass(self):
+        '''Called after tests in class.
+        '''
+        super(BaseTestWithDB, self).setUpClass()
+
+    def setUp(self):
+        '''Called before each test.
+        Sets the language to English, creates a new user and logs into the database
```
Comment (hayleyavw) [inline csunplugged/tests/BaseTestWithDB.py]: does this need to happen before each test? or just once at the start of the test suite execution?

File: csunplugged/tests/topics/test_models.py [hunk 4]
```diff
@@ -1,5 +1,4 @@
-import unittest
```
Comment (hayleyavw) [inline csunplugged/tests/topics/test_models.py]: what is the purpose of this file now?

Comment (hayleyavw) [inline csunplugged/tests/topics/test_models.py]: and for the two below?

File: csunplugged/tests/topics/urls/all_curriculum_integrations.py [hunk 5]
```diff
@@ -1,12 +1,12 @@
-from tests.BaseTest import BaseTest
+from tests.BaseTestWithDB import BaseTestWithDB
```
Comment (hayleyavw) [inline csunplugged/tests/topics/urls/all_curriculum_integrations.py]: why do url tests need the db?

File: csunplugged/tests/BaseTest.py [hunk 6]
```diff
@@ -31,4 +31,4 @@ def tearDown(self):
         '''Called after each test.
         Deletes the user.
         '''
-        self.test_user.delete()
+        pass
```
Comment (hayleyavw) [inline csunplugged/tests/BaseTest.py]: did you mean to leave this method blank? if so, fix up the docstring :)

**Comment by ravenmaster001** [comment]
@hayleyavw should be ready for another review. I took another look at how to setup a single user for the whole class using the `setUpData` method and realised a bug with my `tearDownClass` method.

**Pull request merged**